src/plone/restapi/serializer/site.py

```diff
@@ -121,6 +122,8 @@ def __call__(self, version=None):
             for brain in batch
         ]
 
+        result["allow_discussion"] = get_allow_discussion_value(self.context, self.request, result)
```

Comment:
The assignment is being done inside the function. So we shouldn't do assignment here.

```suggestion
        get_allow_discussion_value(self.context, self.request, result)
```